Userland/Libraries/LibGUI/ListView.h

```diff
@@ -56,11 +57,13 @@ class ListView : public AbstractView {
     Gfx::IntRect content_rect(int row) const;
     int item_count() const;
     void update_content_size();
+    int calculate_content_width() const;
 
     int m_horizontal_padding { 2 };
     int m_model_column { 0 };
     bool m_alternating_row_colors { true };
     bool m_hover_highlighting { false };
+    int m_cached_content_width { -1 };
```

Comment:
-1 is a poor value for a placeholder that's not set. This should probably use Optional<int> .

Reply:
Is this a Serenity-wide rule? I actually quite like -1 as an "invalid" value for an int that won't be negative, just because it's an efficient use of a machine word. I like Optional for things like objects with no "invalid" value.

I do see several uses of -1 like this in Serenity already, so it's probably not too frowned upon?

e.g.
https://github.com/SerenityOS/serenity/blob/a8967388d3b6986dd10fab193c0b1fef287a4e18/Userland/Shell/Builtin.cpp#L111

https://github.com/SerenityOS/serenity/blob/76a18e1172a9389fad540fcfdf81252e237da2de/Userland/Services/WindowServer/WindowSwitcher.cpp#L61

https://github.com/SerenityOS/serenity/blob/631d36fd9821e48b3b3b53ac609e2e50b254d15b/Userland/Games/Hearts/Player.cpp#L77

Reply:
I'm not sure on the logic here. The way I'm looking at this commit, You either have a cached width or you don't. A cached width of size -1 is not valid.

 It's the general design policy of this project that if a construct is more expressive, and makes it clearer to the reader what's happening, then that should be the default tool you reach for. 

I'm not sure that the efficiency of this particular algorithm will be that much affected by an extra boolean attached to the cached width. Though if you have some performance numbers that show this really hot path (?) can't afford an extra bool in the cache line, then we can worry about the data locality efficiency of the list view class.

Reply:
I don't have any performance numbers for this and this isn't a hot path. I just wanted to gently push back on this suggestion because this use of -1 seems quite common in this project. It sounds like you're still quite keen on Optional here, though, so I'm happy to use it.

Reply:
Done! 🙂